Patches.py

```diff
@@ -1000,6 +1000,19 @@ def set_entrance_updates(entrances):
             save_context.write_permanent_flag(Scenes.SHADOW_TEMPLE, FlagType.SWITCH, 0x0, 0x20) # Shadow Boat Block
             save_context.write_permanent_flag(Scenes.SHADOW_TEMPLE, FlagType.SWITCH, 0x1, 0x01) # Shadow Bird Bridge
 
+    if world.settings.plant_beans:
+        save_context.write_permanent_flag(Scenes.GRAVEYARD, FlagType.SWITCH, 0x1, 0x03) # Plant Graveyard bean
+        save_context.write_permanent_flag(Scenes.ZORAS_RIVER, FlagType.SWITCH, 0x1, 0x03) # Plant Zora's River bean
+        save_context.write_permanent_flag(Scenes.KOKIRI_FOREST, FlagType.SWITCH, 0x1, 0x09) # Plant Kokiri Forest bean
+        save_context.write_permanent_flag(Scenes.LAKE_HYLIA, FlagType.SWITCH, 0x1, 0x01) # Plant Lake Hylia bean
+        save_context.write_permanent_flag(Scenes.GERUDO_VALLEY, FlagType.SWITCH, 0x1, 0x03) # Plant Gerudo Valley bean
+        save_context.write_permanent_flag(Scenes.LOST_WOODS, FlagType.SWITCH, 0x1, 0x04) # Plant Lost Woods bridge bean
+        save_context.write_permanent_flag(Scenes.LOST_WOODS, FlagType.SWITCH, 0x1, 0x12) # Plant Lost Woods theater bean
+        save_context.write_permanent_flag(Scenes.DESERT_COLOSSUS, FlagType.SWITCH, 0x1, 0x18) # Plant Desert Colossus bean
+        save_context.write_permanent_flag(Scenes.DEATH_MOUNTAIN_TRAIL, FlagType.SWITCH, 0x1, 0x06) # Plant Death Mountain Trail bean
+        save_context.write_permanent_flag(Scenes.DEATH_MOUNTAIN_CRATER, FlagType.SWITCH, 0x1, 0x03) # Plant Death Mountain Crater bean
+        save_context.write_bits(0x009B, 0x0A) # 10 magic beans sold
+
```

Comment:
I haven't confirmed this works yet.

Reply:
Writing 0x10 to the item slot doesn't seem to be working still (SaveContext.item_id_map['beans']). Unfortunately, `give_item('Magic Beans')` results in 10 beans and 0 beans sold, so that's not going to work.

Reply:
The switch flags work now though.

Reply:
Alright, I finally got this working how I want with `give_item` and setting the `magic_beans_sold` address value. I realized I wasn't setting the value on the address.

Reply:
I still need to test this with beans shuffled

Reply:
Everything looks fine with beans shuffled. Check every bean as child and adult. Ensured I could still plant bugs.